src/mining_pits.cpp

```diff
@@ -0,0 +1,279 @@
+				if (!pit_mask->check_within(neighbor.row,neighbor.col))
+					continue;
+				if ((directions[d].row * directions[d].col == 0) && (!seen->get(neighbor.row,neighbor.col))) {
+					q.push(neighbor);
+				}
+			}
+		}
+	}
+
+	return pit_area;
+}
+
+ArrayCoordinate find_lowest_point_pit_lake(Model<bool> *individual_pit_mask) {
+	// The lowest point is assumed to be at the Point of Inaccessibility (POI) for the pit polygon
+	// For a perfect circle, the POI would be the centre of the circle
+	
+	queue<ArrayCoordinate> q;
+    vector<vector<int>> dist(individual_pit_mask->nrows(), vector<int>(individual_pit_mask->ncols(), INT_MAX));
+    
+    // Push all boundary cells (value 0 in raster) into the queue and set their distance to 0
+    for(int row = 0; row<individual_pit_mask->nrows();row++) {
+		for(int col = 0; col<individual_pit_mask->ncols();col++) {	
+            if (!individual_pit_mask->get(row,col)) {
+				ArrayCoordinate bc = ArrayCoordinate_init(row,col,individual_pit_mask->get_origin());
+                q.push(bc);
+				dist[row][col] = 0;
+            }
+        }
+    }
+
+	// Find distance between internal polygon cells and the polygon boundary
+    while (!q.empty()) {
+        ArrayCoordinate p = q.front();
+        q.pop();
+
+		for (uint d=0; d<directions.size(); d++) {
+			ArrayCoordinate neighbor = ArrayCoordinate_init(p.row + directions[d].row, p.col + directions[d].col, individual_pit_mask->get_origin());
+			if (directions[d].row * directions[d].col != 0)
+				continue;
+			if ((!individual_pit_mask->check_within(neighbor.row,neighbor.col)) || (!individual_pit_mask->get(neighbor.row,neighbor.col))) {
+				continue;
+			}
+
+			if (dist[p.row][p.col] + 1 < dist[neighbor.row][neighbor.col]) {
+				dist[neighbor.row][neighbor.col] = dist[p.row][p.col] + 1;
+				q.push(neighbor);
+			}
+		}
+    }
+	
+	// Determine the Point of Inaccessibility based on the maximum distance between any cell and the polygon boundary
+	int max_distance = -1;
+    ArrayCoordinate lowest_point = {-1, -1, individual_pit_mask->get_origin()};
+    
+    for(int row = 0; row<individual_pit_mask->nrows();row++) {
+		for(int col = 0; col<individual_pit_mask->ncols();col++) {
+            if (dist[row][col] > max_distance) {
+				max_distance = dist[row][col];
+                lowest_point = {row, col, individual_pit_mask->get_origin()};
+            }
+        }
+    }
+
+	return lowest_point;
+}
+
+double find_volume_pit_lake(double pit_area, int pit_depth) {
+	// Pit lake volumes are calculated by assuming that they are conical structures with a flat bottom
+	// Find height of cone with pit_lake surface as base
+	double pit_lake_surface_r = sqrt(pit_area / pi);
+	double pit_lake_bottom_r = sqrt(pit_lake_relative_area * pit_area / pi);
+	double surface_bottom_r_diff = pit_lake_surface_r - pit_lake_bottom_r;
+	double wall_vertical_angle = atan(surface_bottom_r_diff/pit_depth);
```

Comment:
Changed this to just be the volume of a cylinder (rather than assuming a flat bottom as well). Should provide a more reasonable estimate and avoid the issue of excessively steep angles.